airbyte-integrations/bases/source-acceptance-test/source_acceptance_test/tests/test_core.py

```diff
@@ -114,6 +114,21 @@ def test_discover(self, connector_config, docker_runner: ConnectorRunner):
         #         stream2.json_schema = None
         #         assert stream1.dict() == stream2.dict(), f"Streams {stream1.name} and {stream2.name}, stream configs should match"
 
+    def test_defined_cursors_exist_in_schema(self, connector_config, docker_runner: ConnectorRunner):
+        """
+        Check if all of the source defined cursor fields are exists on stream's json schema.
+        """
+        output = docker_runner.call_discover(config=connector_config)
+        catalog = filter_output(output, Type.CATALOG)[0].catalog
+        for stream in catalog.streams:
+            if stream.default_cursor_field:
+                schema = stream.json_schema
+                assert "properties" in schema, "Top level item should have an 'object' type for {stream.name} stream schema"
+                properties = schema["properties"].keys()
+                assert all(
+                    [cursor in properties for cursor in stream.default_cursor_field]
```

Comment:
This won't work with deep level cursor fields.
- Assume we have `cursor_field=["f1", "f2"]`, which is the cursor path specified as a list of path parts.
- So this cursor field may be accessible with `schema["properties"]["fa1"]["f2"]`

Reply:
Got it, updated PR.